website/other/bezier-rs-demos/wasm/src/bezier.rs

```diff
@@ -353,19 +348,53 @@ impl WasmBezier {
 		wrap_svg_tag(content)
 	}
 
-	/// The wrapped return type is `Vec<Vec<Point>>`.
-	pub fn de_casteljau_points(&self, t: f64) -> JsValue {
-		let points: Vec<Vec<Point>> = self
-			.0
-			.de_casteljau_points(t)
+	pub fn de_casteljau_points(&self, t: f64) -> String {
+		let points: Vec<Vec<DVec2>> = self.0.de_casteljau_points(t);
+
+		let bezier_svg = self.get_bezier_path();
+
+		let casteljau_svg = points
 			.iter()
-			.map(|level| level.iter().map(|&point| Point { x: point.x, y: point.y }).collect::<Vec<Point>>())
-			.collect();
-		to_js_value(points)
+			.enumerate()
+			.map(|(idx, points)| {
+				let color_light = format!("hsl({}, 100%, 50%)", 90 * idx);
+				let points_and_handle_lines = points
+					.iter()
+					.enumerate()
+					.map(|(index, point)| {
+						format!(
+							"{}{}",
+							draw_circle(point.x, point.y, 3., &color_light, 1.5, WHITE),
+							if index != 0 {
+								let prev_point = points[index - 1];
+								draw_line(prev_point.x, prev_point.y, point.x, point.y, &color_light, 1.5)
+							} else {
+								String::new()
+							}
+						)
+					})
+					.fold("".to_string(), |acc, point_svg| acc + &point_svg);
```

Comment:
Use `String::new()` for consistency and unabbreviate `acc` here and for the line below. Alternatively, could we just use `.concat()`?

Reply:
I think it makes sense to use this because it is more consistent with the existing code (i.e. local extrema, inflections, compute lookup table)